### PR TITLE
Modify judgement of placement changed and consider clusteraffinities

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #3153 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/helper.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	"encoding/json"
+	"reflect"
+
+	"k8s.io/klog/v2"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+)
+
+func needConsideredPlacementChanged(
```
Comment (RainbowMango) [inline pkg/scheduler/helper.go:12]:
```suggestion
func placementChanged(
```
we don't need to check `if we need consider`, we definitely are considering if placement changes.

File: pkg/scheduler/helper.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	// third check: check weather ClusterAffinities has changed
+	return clusterAffinitiesChanged(placement.ClusterAffinities, appliedPlacement.ClusterAffinities, schedulerObservingAffinityName)
+}
+
+func clusterAffinitiesChanged(
+	clusterAffinities, appliedClusterAffinities []policyv1alpha1.ClusterAffinityTerm,
+	schedulerObservingAffinityName string,
+) bool {
+	// indicates that scheduling has not performed and needs to be performed
```
Comment (RainbowMango) [inline pkg/scheduler/helper.go:49]: This looks suspicious, from this function's point of view, it just needs to care about if a particular affinity term changes.

Comment (XiShanYongYe-Chang) [reply pkg/scheduler/helper.go]: I remove this comment.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**